### ISSUE TITLE
clang -Wdocumentation is too picky about empty paragraph

### ISSUE THREAD

**Issue by gilles-peskine-arm** [llvm/llvm-project#60315]
Clang's `-Wdocumentation` diagnostic rejects `\param <name> {description}`, `\return {description}` and `\retval <value> {description}` (possibly more) when the `{description}` part is empty. However, the [Doxygen documentation](https://doxygen.nl/manual/commands.html#cmdretval) does not suggest that the `{description}` must not be empty, and the `doxygen` command does not complain.

I can see the point that empty documentation is not as good as non-empty documentation. However, [in the real world](https://github.com/Mbed-TLS/mbedtls/issues/6960), we have a code base with many `\retval ERROR_CODE` annotations with an empty description, where the details of the meaning of the error code in each function is not particularly important. Clang started to complain about this in version 15.0.0, so Clang 15 broke our build.

Small reproducer: `a.h`
```
/** \file a.h */

/** Do stuff.
 *
 * \param n
 * \return
 * \retval 42
 */
int foo(int n);
```

Compilation with clang 15.0.6 or `16.0.0 (++20230101052931+d408c34d1f58-1~exp1~20230101173033.495` (from http://apt.llvm.org/focal/):
```
clang-15 -Wdocumentation -c a.h
a.h:5:11: warning: empty paragraph passed to '\param' command [-Wdocumentation]
 * \param n
   ~~~~~~~^
a.h:6:11: warning: empty paragraph passed to '\return' command [-Wdocumentation]
 * \return
   ~~~~~~~^
a.h:7:13: warning: empty paragraph passed to '\retval' command [-Wdocumentation]
 * \retval 42
   ~~~~~~~~~^
3 warnings generated.
```

Clang 10.x through 14.x only warns about `\param` and `\return`, not `\retval`. Clang ≤9.x only warns about `\param`.

Desired behavior: `-Wdocumentation` does not result in warnings about empty paragraphs, at least not with `\retval`.
